src/spaces/compartment.jl

```diff
@@ -1,29 +1,25 @@
 export CompartmentSpace
 
-struct CompartmentSpace{D,F} <: AbstractSpace
-  grid::GridSpace
+struct CompartmentSpace{G,F,D} <: AbstractSpace
+  grid::G
```

Comment:
I can see from the definition of a GridSpace that https://github.com/JuliaDynamics/Agents.jl/blob/master/src/spaces/grid.jl#L19  `D, P` are the parameters of the grid space. **These** should be made type parameters of the CompartmentSpace, whose field should then type `::GridSpace{D, P}`. Also, this doesn't duplicate `D`.